zjit/src/backend/arm64/mod.rs

```diff
@@ -706,70 +706,50 @@ impl Assembler
         /// Emit a conditional jump instruction to a specific target. This is
         /// called when lowering any of the conditional jump instructions.
         fn emit_conditional_jump<const CONDITION: u8>(cb: &mut CodeBlock, target: Target) {
+            /// Helper method to generate a branch instruction.
+            fn generate_branch<const CONDITION: u8>(cb: &mut CodeBlock, offset: InstructionOffset, dst_addr: i64) {
+                let num_insns = if bcond_offset_fits_bits(offset.into()) {
+                    bcond(cb, CONDITION, offset);
+                    1
+                } else if b_offset_fits_bits(offset.into()) {
+                    // Since we cannot encode the offset in a bcond directly, generate
+                    // an unconditional branch and then jump past it only if the inverse
+                    // of the original condition is met.
+                    bcond(cb, Condition::inverse(CONDITION), 2.into());
+                    b(cb, offset);
+                    2
+                } else {
+                    // Since we cannot encode the offset into an instruction, load it into
+                    // some number of registers and then call br.
+                    // NOTE: I'm skeptical of the `.try_into().unwrap()`
```

Comment:
It's because it's wrong. For cases where it'd panic (negative numbers) we actually always want to always preserve the bit string of the address. So this should always use a `i64 as u64` cast.

Can we take old `CodePtr` version instead of this rewrite? You also have some legit CI crashes.